src/server/components/charts-engine/components/processor/index.ts

```diff
@@ -1084,6 +1085,12 @@ export class Processor {
                         statusCode: DEFAULT_RUNTIME_TIMEOUT_STATUS,
                     };
 
+                    onCodeExecuted({
```

Comment:
This will be run on timeout on any tab, right? Probably only needed for js?

Reply:
Yes, but this is normal, because we are interested in all cases of timeouts, and the separation is difficult. But this can be changed after refactoring.